connectors/protocol/connectors.py

```diff
@@ -662,22 +673,38 @@ async def error(self, error):
     async def sync_done(self, job, cursor=None):
         job_status = JobStatus.ERROR if job is None else job.status
         job_error = JOB_NOT_FOUND_ERROR if job is None else job.error
+        job_type = job.job_type if job is not None else None
+
         if job_error is None and job_status == JobStatus.ERROR:
             job_error = UNKNOWN_ERROR
         connector_status = (
             Status.ERROR if job_status == JobStatus.ERROR else Status.CONNECTED
         )
 
+        last_sync_information = (
+            {
+                "last_access_control_sync_status": job_status.value,
+                "last_access_control_sync_error": job_error,
+            }
+            if job_type == JobType.ACCESS_CONTROL
+            else {
+                "last_sync_status": job_status.value,
+                "last_sync_error": job_error,
+            }
+        )
+
         doc = {
-            "last_sync_status": job_status.value,
             "last_synced": iso_utc(),
-            "last_sync_error": job_error,
             "status": connector_status.value,
             "error": job_error,
-        }
+        } | last_sync_information
 
         # only update sync cursor after a successful sync job
-        if job_status == JobStatus.COMPLETED:
+        if (
+            job_type is not None
```

Comment:
`is not None` is a double negative. I'd prefer `job_type in [JobType.FULL, JobType.INCREMENTAL]` to make the logic a little more explicit and clear.